### PR TITLE
Add two new `ArgumentArrayParsingStrategy` options

### PR DESCRIPTION
**Pull request by natecook1000**
### Description
This adds two new parsing options for argument arrays, and renames `.unconditionalRemaining` to `.captureForPassthrough`:

- `unconditionalRemaining` has been an attractive nuisance. It looks like it means "everything leftover", but it actually has a specific and slightly strange set of behaviors to support SwiftPM capturing inputs to pass on to a sub-client, as in `swift run MyExecutableTarget`. The new name better matches this intended usage.
- `allUnrecognized` collects all the inputs that weren't used during parsing, which is a more generally useful behavior. This essentially suppresses all "unrecognized flag/option" and "unexpected argument" errors, and makes those extra inputs available to the client.
- `postTerminator` collects all inputs that follow the `--` terminator, before trying to parse any other positional arguments. This is a non-standard, but sometimes useful parsing strategy.

### Detailed Design
Two new symbols and one renaming — see the diff for full API documentation.

```swift
/// After parsing, capture all unrecognized inputs in this argument array.
public static var allUnrecognized: ArgumentArrayParsingStrategy {
  self.init(base: .allUnrecognized)
}

/// Before parsing, capture all inputs that follow the `--` terminator in this
/// argument array.
public static var postTerminator: ArgumentArrayParsingStrategy {
  self.init(base: .postTerminator)
}

/// Parse all remaining inputs after parsing any known options or flags,
/// including dash-prefixed inputs and the `--` terminator.
public static var captureForPassthrough: ArgumentArrayParsingStrategy {
  self.init(base: .allRemainingInput)
}

@available(*, deprecated, renamed: "captureForPassthrough")
public static var unconditionalRemaining: ArgumentArrayParsingStrategy {
  .captureForPassthrough
}
```

### Documentation Plan
Includes API documentation for the new symbols and significantly expanded docs for the existing members.

### Test Plan
Includes unit tests for the new cases.

### Source Impact
This deprecates the `unconditionalRemaining` strategy, which was confusingly named. The new name is `captureForPassthrough`, which is more explicit about its intent.

### Checklist
- [x] I've added at least one test that validates that my change is working, if appropriate
- [x] I've followed the code style of the rest of the project
- [x] I've read the [Contribution Guidelines](https://github.com/apple/swift-argument-parser/blob/main/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please test

File: Sources/ArgumentParser/Parsing/ArgumentDefinition.swift [hunk 1]
```diff
@@ -91,6 +91,11 @@ struct ArgumentDefinition {
     case upToNextOption
     /// Parse all remaining `SplitArguments.Element` as values, regardless of its type.
     case allRemainingInput
+    /// Collect all the elements after the terminator, preventing them from
+    /// appearing in any other position.
+    case postTerminator
+    /// Collect all unused inputs after the
```
Comment (rauhul) [inline Sources/ArgumentParser/Parsing/ArgumentDefinition.swift:97]: after the what!!

**Comment by natecook1000** [comment]
@swift-ci Please test

**Comment by danpalmer** [comment]
@natecook1000 Thanks for implementing this!

**Pull request merged**